lib/charted.js

```diff
@@ -9,16 +9,19 @@ module.exports = function(app, charted_root) {
   if (charted_root === undefined) { charted_root = '/charted/' }
 
   app.use(charted_root + 'get', function (req, res, next) {
-    var get_url = url.parse(req.url, true).query.url
+    var urlObj = url.parse(req.url, true)
 
-    if (!get_url) {
+    if (!urlObj.query.url) {
       res.statusCode = 400
       res.end('Bad Request: no url provided')
-      next()
       return
     }
 
-    request(decodeURIComponent(get_url), function (err, resp, body) {
+    var get_url = urlObj.query.url.match(/:\/\//) ? urlObj.query.url :
```

Comment:
Since `urlObj` is a result of `url.parse`, don't you have all the information in the object already? For example, instead of matching on the url, you can check whether `hostname` is `null`.